pgcli/main.py

```diff
@@ -316,9 +316,23 @@ def fixup_possible_percent_encoding(s):
         # unquote str(each URI part (they may be percent encoded)
         self.connect(**arguments)
 
+    def guess_socketdir(self):
+        candidates = [
+            '/var/run/postgresql',
+            '/var/pgsql_socket',
+            '/usr/local/var/postgres',
+            '/tmp',
+        ]
+
+        for candidate in candidates:
+            if os.path.isdir(candidate):
+                return candidate
+
     def connect(self, database='', host='', user='', port='', passwd='',
                 dsn='', **kwargs):
         # Connect to the database.
+        if not host:
+            host = self.guess_socketdir() or ''
```

Comment:
Actually, I do not like mixing param guess inside `connect` function. This is misleading. I suggest writing a `guess_dsn` function that do all the guess I/O, env var fallback, etc. and let `connect` just trigger the connection or fail.